astrokat/noisediode.py

```diff
@@ -0,0 +1,67 @@
+import time
+
+from simulate import user_logger
+
+
+# switch noise-source pattern off
+def on(mkat, logging=True):
+    if logging:
+        user_logger.info('Switch noise-diode on')
+    # Noise Diodes are triggered on all antennas in array simultaneously
+    # add a second to ensure all digitisers set at the same time
+    timestamp = time.time() + 1
+    mkat.ants.req.dig_noise_source(timestamp, 1)
+
+
+# switch noise-source pattern off
+def off(mkat, logging=True):
+    if logging:
+        user_logger.info('Switch noise-diode off')
+    # Noise Diodes are triggered on all antennas in array simultaneously
+    # add a second to ensure all digitisers set at the same time
+    timestamp = time.time() + 1
+    mkat.ants.req.dig_noise_source(timestamp, 0)
+
+
+# fire noisediode before track
```

Comment:
Please be consistent - some comments refer to "noise-source", some to "noisediode".
I like "noise diode".